client/src/docToPreviewGenerator.ts

```diff
@@ -89,7 +89,13 @@ export class DocToPreviewGenerator {
 
         let list = "";
         arr.forEach((s) => {
-          list += `<li>${s}</li>`;
+          if (s.length > 0) {
+            const rg = new RegExp(/^(.*?):(\d+):(\d+):(\s+)(.*)$/g).exec(s);
```

Comment:
I think you can create the regexp once outside and use the same one in the forEach
Adding a comment with an example match would be helpful

Reply:
For readability I'd prefer to keep this all together, this loop will be execute 3 or less times in the 99.9% case.  Performance is not an issue here. I did add a couple of examples.